### ISSUE TITLE
Whether it is suitable for deployment on a personal computer

### ISSUE THREAD

**Issue by Amphetaminewei** [qdrant/qdrant#4930]
We wanted to build an AIPC, so we were looking for a vector database that could be deployed on the user's personal computer to manage the user's data. In this scenario, there may be tens of millions of vectors. For this use case, there may be some special requirements, such as a memory management mechanism that does not affect the normal use of users, and cannot load indexes in memory for a long time for retrieval speed, as in the cloud.

Is qdrant suitable for this scenario?

**Comment by generall** [comment]
I can't say it was built for that. Especially if te user's computer runs on windows - there might be some performance limitation compare to linux-based deployments. Overall I would recommend to try to host qdrant at least in local network rather than the consumer laptop.

**Comment by timvisee** [comment]
But, there is no technical limitation that would prevent you from doing this.

**Comment by Amphetaminewei** [comment]
@generall  Thank you for your reply. The target of our deployment is a linux-based desktop operating system, and I see that there is a deb package in the release, which is exactly what we need. We want to provide users with a feature that allows them to find documents locally through similar searches.

I would like to ask what is the memory usage strategy of qdrant server, and will it continue to keep all indexes in memory like Chroma server for subsequent retrieval? For us, if a vector retrieval is not used for a long time and the memory is not freed, the user's computer may become stuck.

**Comment by timvisee** [comment]
> I would like to ask what is the memory usage strategy of qdrant server, and will it continue to keep all indexes in memory like Chroma server for subsequent retrieval? For us, if a vector retrieval is not used for a long time and the memory is not freed, the user's computer may become stuck.

The strategy is basically, use as much memory as possible.

You can use `on_disk: true` flags which will make use of memory mapping. That'll allow the Linux kernel to release pages as needed, to keep your system operational. But I'm not sure if that is sufficient.

I'd recommend to test with this to see if it meets your requirements.

**Comment by Amphetaminewei** [comment]
@timvisee Yes, we will do some trial and research to see if qdrant is suitable for our scenario

**Comment by Amphetaminewei** [comment]
> I'd recommend to test with this to see if it meets your requirements.

@timvisee  Okay, we'll go test and try on_disk, thank you
